exporter/exporterhelper/internal/queue_sender.go

```diff
@@ -73,6 +73,8 @@ type QueueSender struct {
 	traceAttribute attribute.KeyValue
 	consumers      *queue.Consumers[internal.Request]
 
+	shutdownCallbacks []func()
```

Comment:
I was actually looking for an established pattern, but didn't find one. Can you point me to it? I searched for io.Closer in the code, and found only one reference which is unrelated.

Reply:
Interestingly, I wasn't able to find one example either in this repo :-) There is exactly one usage of this pattern in contrib.

We do have a `func Shutdown(context.Context) error` as part of our `component.Component` interface, but I'd argue in favor of io.Closer in this case.

Reply:
This was the only thing I found that was in the same neighborhood: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/logstransformprocessor/processor.go#L66

Reply:
So where can I actually see that io.Closer use? I found the component.ShutdownFunc thing, but not the other one

Reply:
I misread the code in contrib, it was only a type check 🤦🏽 I scratched something here, but I think I prefer your approach, perhaps with the only improvement that it could return an error, to be logged on your iterator.

Reply:
I'm now logging (warn level) if there's any error unregistering the callbacks.